src/components/date-picker/date-picker.tsx

```diff
@@ -57,9 +84,7 @@ export class DatePicker extends React.Component<DatePickerProps, DatePickerState
             const updatedDate = input ? new Date(input) : new Date();
             this.setState({ inputValue: updatedDate.toDateString() });
 
-            if (this.props.onChange) {
-                this.props.onChange(updatedDate);
-            }
+            this.props.onChange!(updatedDate);
```

Comment:
I wouldn't expect a call to `props.onChange` to be inside a method named `updateState*`.  a bit confusing and breaks the single responsibility principle. Extract it to a more properly named method.